runtime/common/src/migrations/precompile_account_codes.rs

```diff
@@ -0,0 +1,338 @@
+				H160::from(crate::evm::precompile::BN128PAIRING_ADDR),
+				PRECOMPILE_CODE_STORAGE.to_vec(),
+			);
+		} else {
+			log::warn!(
+				"precompile::AccountCodes: BN128PAIRING storage already populated. Skipping."
+			)
+		}
+
+		if pallet_evm::AccountCodes::<T>::get(H160::from(crate::evm::precompile::BLAKE2F_ADDR))
+			.is_empty()
+		{
+			log::info!("precompile::AccountCodes: Inserting code for BLAKE2F.");
+			pallet_evm::AccountCodes::<T>::insert(
+				H160::from(crate::evm::precompile::BLAKE2F_ADDR),
+				PRECOMPILE_CODE_STORAGE.to_vec(),
+			);
+		} else {
+			log::warn!("precompile::AccountCodes: BLAKE2F storage already populated. Skipping.")
+		}
+
+		if pallet_evm::AccountCodes::<T>::get(H160::from(crate::evm::precompile::SHA3FIPS256_ADDR))
+			.is_empty()
+		{
+			log::info!("precompile::AccountCodes: Inserting code for SHA3FIPS256.");
+			pallet_evm::AccountCodes::<T>::insert(
+				H160::from(crate::evm::precompile::SHA3FIPS256_ADDR),
+				PRECOMPILE_CODE_STORAGE.to_vec(),
+			);
+		} else {
+			log::warn!("precompile::AccountCodes: SHA3FIPS256 storage already populated. Skipping.")
+		}
+
+		if pallet_evm::AccountCodes::<T>::get(H160::from(crate::evm::precompile::DISPATCH_ADDR))
+			.is_empty()
+		{
+			log::info!("precompile::AccountCodes: Inserting code for DISPATCH.");
+			pallet_evm::AccountCodes::<T>::insert(
+				H160::from(crate::evm::precompile::DISPATCH_ADDR),
+				PRECOMPILE_CODE_STORAGE.to_vec(),
+			);
+		} else {
+			log::warn!("precompile::AccountCodes: DISPATCH storage already populated. Skipping.")
+		}
+
+		if pallet_evm::AccountCodes::<T>::get(H160::from(
+			crate::evm::precompile::ECRECOVERPUBLICKEY_ADDR,
+		))
+		.is_empty()
+		{
+			log::info!("precompile::AccountCodes: Inserting code for ECRECOVERPUBLICKEY.");
+			pallet_evm::AccountCodes::<T>::insert(
+				H160::from(crate::evm::precompile::ECRECOVERPUBLICKEY_ADDR),
+				PRECOMPILE_CODE_STORAGE.to_vec(),
+			);
+		} else {
+			log::warn!(
+				"precompile::AccountCodes: ECRECOVERPUBLICKEY storage already populated. Skipping."
+			)
+		}
+
+		if pallet_evm::AccountCodes::<T>::get(H160::from(crate::evm::precompile::LP_AXELAR_GATEWAY))
+			.is_empty()
+		{
+			log::info!("precompile::AccountCodes: Inserting code for LP_AXELAR_GATEWAY.");
+			pallet_evm::AccountCodes::<T>::insert(
+				H160::from(crate::evm::precompile::LP_AXELAR_GATEWAY),
+				PRECOMPILE_CODE_STORAGE.to_vec(),
+			);
+		} else {
+			log::warn!(
+				"precompile::AccountCodes: LP_AXELAR_GATEWAY storage already populated. Skipping."
+			)
+		}
+
+		log::info!("precompile::AccountCodes: Inserting precompile account codes, on_runtime_upgrade: completed!");
+
+		// NOTE: This is a worst case weight and we do not care to adjust it correctly
+		// depending on skipped read/writes.
+		Weight::from_ref_time(
+			T::DbWeight::get()
+				.read
+				.ensure_mul(13)
+				.unwrap_or(u64::MAX)
+				.ensure_add(T::DbWeight::get().write.ensure_mul(13).unwrap_or(u64::MAX))
+				.unwrap_or(u64::MAX),
+		)
```

Comment:
I would love this to be simplified without the need to unwrap:

```suggestion
		Weight::from_ref_time(
			T::DbWeight::get()
				.reads_writes(13, 13)
		)
```